endpoints/info/bidders.go

```diff
@@ -4,28 +4,69 @@ import (
 	"encoding/json"
 	"net/http"
 	"sort"
+	"strings"
 
 	"github.com/golang/glog"
 	"github.com/julienschmidt/httprouter"
 	"github.com/prebid/prebid-server/config"
 )
 
+var invalidEnabledOnly = []byte(`Invalid value for 'enabledonly' query param, must be of boolean type`)
+
 // NewBiddersEndpoint builds a handler for the /info/bidders endpoint.
 func NewBiddersEndpoint(bidders config.BidderInfos, aliases map[string]string) httprouter.Handle {
-	response, err := prepareBiddersResponse(bidders, aliases)
+	responseAll, err := prepareBiddersResponseAll(bidders, aliases)
+	if err != nil {
+		glog.Fatalf("error creating /info/bidders endpoint all bidders response: %v", err)
+	}
+
+	responseEnabledOnly, err := prepareBiddersResponseEnabledOnly(bidders, aliases)
 	if err != nil {
-		glog.Fatalf("error creating /info/bidders endpoint response: %v", err)
+		glog.Fatalf("error creating /info/bidders endpoint enabled only response: %v", err)
 	}
 
-	return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
-		w.Header().Set("Content-Type", "application/json")
-		if _, err := w.Write(response); err != nil {
-			glog.Errorf("error writing response to /info/bidders: %v", err)
+	return func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+		var writeErr error
+
+		enabledOnly, exists := readEnabledOnly(r)
+
+		if !exists {
+			// Backwards compatibility. PBS would previously respond with all enabled bidders.
+			w.Header().Set("Content-Type", "application/json")
+			_, writeErr = w.Write(responseAll)
+		} else {
+			switch enabledOnly {
```

Comment:
Thoughts about extracting this logic in a diff function? Perhaps we can call it something like `checkEnabledOnly` which takes in a request and returns a bool value `enabledOnly` and an error. It will call the `readEnabledOnly` function and return `true` if the `enabledOnly` query parameter exists and is set to `true`, `false` if it either doesn't exist or is explicitly set to `false`, error otherwise. With that the logic in this handler function will become much more sleek and easier to follow. It would look something along the lines of:
```
enabledOnly, err := checkEnabledOnly(r)
if(err != nil) {
	w.WriteHeader(http.StatusBadRequest)
	_, writeErr = w.Write(invalidEnabledOnly)
}

w.Header().Set("Content-Type", "application/json")
if(enabledOnly) {
	_, writeErr = w.Write(responseEnabledOnly)
} else {
	_, writeErr = w.Write(responseAll)
}
```